### PR TITLE
Bravo 1881 bulk purchase

### PR DESCRIPTION
**Pull request by SKonstantinS**
Ask for review
But I have to decrease gas consumption for bulk transfer Wyvern orders

### CODE REVIEW

File: exchange-v2/contracts/ExchangeBulkV2.sol [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import "./ExchangeV2Core.sol";
+import "./RaribleTransferManager.sol";
+import "@rarible/royalties/contracts/IRoyaltiesProvider.sol";
+import "@rarible/exchange-interfaces/contracts/IWyvernExchange.sol";
+
+contract ExchangeBulkV2 is ExchangeV2Core, RaribleTransferManager {
```
Comment (disenotov) [inline exchange-v2/contracts/ExchangeBulkV2.sol:11]: why is it inherited from RTM and exchangeV2?

File: exchange-v2/contracts/ExchangeBulkV2.sol [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    ) external payable {
+        uint totalAmount;
+        for (uint256 i = 0; i < sellOrders.length; i++) {
+            validateFull(sellOrders[i], sellOrderSignatures[i]);
+            LibOrder.Order memory buyerOrder;
+            buyerOrder.maker = buyer;
+            buyerOrder.makeAsset = sellOrders[i].takeAsset;
+            buyerOrder.takeAsset = sellOrders[i].makeAsset;
+            buyerOrder.dataType = 0xffffffff;
+            totalAmount += matchAndTransferBulk(sellOrders[i], buyerOrder);
```
Comment (disenotov) [inline exchange-v2/contracts/ExchangeBulkV2.sol:72]: why do we need additional method?

File: exchange-v2/contracts/ExchangeBulkV2.sol [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        }
+        if (msg.value > totalAmount) {
+            address(_msgSender()).transferEth(msg.value.sub(totalAmount));
```
Comment (disenotov) [inline exchange-v2/contracts/ExchangeBulkV2.sol:75]: if returning change is a problem we can probably skip it

File: exchange-v2/contracts/ExchangeBulkV2.sol [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        }
+    }
+
+    /*
+    */
+    function matchWyvernExchange(
+        WyvernOrder memory order,
+        uint ethAmount
+    ) internal {
+        WyvernOrders memory wyvernOrders;
```
Comment (disenotov) [inline exchange-v2/contracts/ExchangeBulkV2.sol:120]: why do we need this variable? can't we form the data at the time of calling atomicMatch_() ?

File: exchange-v2/contracts/ExchangeBulkV2.sol (hunk 1 above)
Comment (SKonstantinS) [reply exchange-v2/contracts/ExchangeBulkV2.sol]: **done**, delete inheritance

File: exchange-v2/contracts/ExchangeBulkV2.sol (hunk 2 above)
Comment (SKonstantinS) [reply exchange-v2/contracts/ExchangeBulkV2.sol]: done, delete it

File: exchange-v2/contracts/ExchangeBulkV2.sol (hunk 3 above)
Comment (SKonstantinS) [reply exchange-v2/contracts/ExchangeBulkV2.sol]: I think it`s not a problem

Comment (SKonstantinS) [reply exchange-v2/contracts/ExchangeBulkV2.sol]: so return change works good

File: exchange-v2/contracts/ExchangeBulkV2.sol (hunk 4 above)
Comment (SKonstantinS) [reply exchange-v2/contracts/ExchangeBulkV2.sol]: done, I don`t this variable now

**Comment by SKonstantinS** [comment]
close, because not actual, plus all functionality(bulk transaction) is supported in 
https://github.com/rarible/protocol-contracts/pull/144

**Pull request closed without merging**